In [0]:
#mounting Azure Data Lake Storage Account onto DBFS:
configs = {"fs.azure.account.auth.type": "OAuth",
"fs.azure.account.oauth.provider.type": "org.apache.hadoop.fs.azurebfs.oauth2.ClientCredsTokenProvider",
"fs.azure.account.oauth2.client.id": "c17f38ea-3086-4232-ae22-1abcc6e55816",
"fs.azure.account.oauth2.client.secret": "fYm8Q~LNSbsNg7.sC.A2ooHFp_xP_bfbTq_OAaSA",
"fs.azure.account.oauth2.client.endpoint": "https://login.microsoftonline.com/ab817701-0a9c-4e3f-b76c-7ab44709dbe2/oauth2/token"}

dbutils.fs.mount(
source = "abfss://tokyo-olympics-data@tokyoolympicdatans.dfs.core.windows.net", #storage_container@storage_account
mount_point = "/mnt/tokyoolympic",
extra_configs = configs)
  

In [0]:
%fs
ls "mnt/tokyoolympic"

path name size modificationTime dbfs:/mnt/tokyoolympic/raw-data/ raw-data/ 0 1738109377000 dbfs:/mnt/tokyoolympic/transformed-data/ transformed-data/ 0 1738109398000

In [0]:
spark

SparkSession - hive 
 
 
 SparkContext 

 Spark UI 

 
 Version 
 v3.3.2 
 Master 
 local[*, 4] 
 AppName 
 Databricks Shell

In [0]:
athletes = spark.read.format("csv").option("header", "true").load("/mnt/tokyoolympic/raw-data/athletes.csv") 
coaches = spark.read.format("csv").option("header", "true").load("/mnt/tokyoolympic/raw-data/coaches.csv") 
entries_gender = spark.read.format("csv").option("header", "true").option("inferSchema", "true").load("/mnt/tokyoolympic/raw-data/entriesgender.csv") 
medals = spark.read.format("csv").option("header", "true").option("inferSchema", "true").load("/mnt/tokyoolympic/raw-data/medals.csv") 
teams = spark.read.format("csv").option("header", "true").load("/mnt/tokyoolympic/raw-data/teams.csv") 


In [0]:
#Verifying dataframe properties
print(athletes)
print(coaches)
print(entries_gender)
print(medals)
print(teams)

DataFrame[PersonName: string, Country: string, Discipline: string]
DataFrame[Name: string, Country: string, Discipline: string, Event: string]
DataFrame[Discipline: string, Female: int, Male: int, Total: int]
DataFrame[Rank: int, TeamCountry: string, Gold: int, Silver: int, Bronze: int, Total: int, Rank by Total: int]
DataFrame[TeamName: string, Discipline: string, Country: string, Event: string]


In [0]:
print("Athletes dataframe:")
athletes.printSchema()

print("Coaches dataframe:")
coaches.printSchema()

print("EntriesGender dataframe:")
entries_gender.printSchema()

print("Medals dataframe:")
medals.printSchema()

print("Teams dataframe:")
teams.printSchema()


Athletes dataframe:
root
 |-- PersonName: string (nullable = true)
 |-- Country: string (nullable = true)
 |-- Discipline: string (nullable = true)

Coaches dataframe:
root
 |-- Name: string (nullable = true)
 |-- Country: string (nullable = true)
 |-- Discipline: string (nullable = true)
 |-- Event: string (nullable = true)

EntriesGender dataframe:
root
 |-- Discipline: string (nullable = true)
 |-- Female: integer (nullable = true)
 |-- Male: integer (nullable = true)
 |-- Total: integer (nullable = true)

Medals dataframe:
root
 |-- Rank: integer (nullable = true)
 |-- TeamCountry: string (nullable = true)
 |-- Gold: integer (nullable = true)
 |-- Silver: integer (nullable = true)
 |-- Bronze: integer (nullable = true)
 |-- Total: integer (nullable = true)
 |-- Rank by Total: integer (nullable = true)

Teams dataframe:
root
 |-- TeamName: string (nullable = true)
 |-- Discipline: string (nullable = true)
 |-- Country: string (nullable = true)
 |-- Event: string (nullable = true)



In [0]:
#Gathering and displaying information about each table.
print("Athletes")
athletes.show()

print("Coaches")
coaches.show()

print("Entries Gender")
entries_gender.show()

print("Medals")
medals.show()

print("Teams")
teams.show()



Athletes
+--------------------+--------------------+-------------------+
|          PersonName|             Country|         Discipline|
+--------------------+--------------------+-------------------+
|     AALERUD Katrine|              Norway|       Cycling Road|
|         ABAD Nestor|               Spain|Artistic Gymnastics|
|   ABAGNALE Giovanni|               Italy|             Rowing|
|      ABALDE Alberto|               Spain|         Basketball|
|       ABALDE Tamara|               Spain|         Basketball|
|           ABALO Luc|              France|           Handball|
|        ABAROA Cesar|               Chile|             Rowing|
|       ABASS Abobakr|               Sudan|           Swimming|
|    ABBASALI Hamideh|Islamic Republic ...|             Karate|
|       ABBASOV Islam|          Azerbaijan|          Wrestling|
|        ABBINGH Lois|         Netherlands|           Handball|
|         ABBOT Emily|           Australia|Rhythmic Gymnastics|
|       ABBOTT Monica|United St

# Exploratory Data Analysis 
Writing various functions to gather insights on the datasets of Athletes, Genders, Teams and Medals.

In [0]:
# Gathering information on total number of athletes, number of disciplines, and number of teams in the Olympics.
print("Total number of athletes: " + str(athletes.count())) 
print("Total number of disciplines: " + str(athletes.select("Discipline").distinct().count()))
print("Total number of teams: " + str(teams.count()))

Total number of athletes: 11085
Total number of disciplines: 46
Total number of teams: 743


### Athletes Data

In [0]:
# Finding Canadian Athletes and their coaches that participated in Basketball events
basketball = athletes.join(coaches, athletes.Discipline == coaches.Discipline, "left") \
                    .filter(athletes.Discipline == "Basketball") \
                    .filter(athletes.Country == "Canada") \
                    .filter(coaches.Country == "Canada") \
                    .withColumnRenamed("PersonName","Athlete") \
                    .withColumnRenamed("Name","Coach").show(100)

+--------------------+-------+----------+--------------+-------+----------+-----+
|             Athlete|Country|Discipline|         Coach|Country|Discipline|Event|
+--------------------+-------+----------+--------------+-------+----------+-----+
|     ACHONWA Natalie| Canada|Basketball|THOMAIDIS Lisa| Canada|Basketball|Women|
|     ALEXANDER Kayla| Canada|Basketball|THOMAIDIS Lisa| Canada|Basketball|Women|
|    AMIHERE Laeticia| Canada|Basketball|THOMAIDIS Lisa| Canada|Basketball|Women|
|        AYIM Miranda| Canada|Basketball|THOMAIDIS Lisa| Canada|Basketball|Women|
|    CARLETON Bridget| Canada|Basketball|THOMAIDIS Lisa| Canada|Basketball|Women|
|         COLLEY Shay| Canada|Basketball|THOMAIDIS Lisa| Canada|Basketball|Women|
|     EDWARDS Aaliyah| Canada|Basketball|THOMAIDIS Lisa| Canada|Basketball|Women|
|        FIELDS Nirra| Canada|Basketball|THOMAIDIS Lisa| Canada|Basketball|Women|
|         GAUCHER Kim| Canada|Basketball|THOMAIDIS Lisa| Canada|Basketball|Women|
|           NURS

In [0]:
# find popularity of Sport by country from Athletes data
sport_popularity = athletes.groupBy("Country","Discipline") \
                            .count() \
                            .orderBy(["count"], ascending=[False])
sport_popularity.show(500)


+--------------------+--------------------+-----+
|             Country|          Discipline|count|
+--------------------+--------------------+-----+
|United States of ...|           Athletics|  144|
|             Germany|           Athletics|   95|
|       Great Britain|           Athletics|   75|
|               Italy|           Athletics|   73|
|               Japan|           Athletics|   70|
|              France|           Athletics|   66|
|           Australia|           Athletics|   65|
|              Poland|           Athletics|   64|
|               Spain|           Athletics|   58|
|              Canada|           Athletics|   56|
|             Jamaica|           Athletics|   54|
|              Brazil|           Athletics|   53|
|People's Republic...|           Athletics|   53|
|             Ukraine|           Athletics|   48|
|         Netherlands|           Athletics|   44|
|           Australia|            Football|   43|
|               Japan|            Football|   42|


In [0]:

# Max number of Athletes from country
from pyspark.sql.functions import max, min

max_count_df = athletes.select("Country") \
    .groupBy("Country") \
    .count() \
    .agg(max("count").alias("max_count")).show()

min_count_df = athletes.select("Country") \
    .groupBy("Country") \
    .count() \
    .agg(min("count").alias("min_count")).show()

+---------+
|max_count|
+---------+
|      615|
+---------+

+---------+
|min_count|
+---------+
|        2|
+---------+



In [0]:
# Getting top 5 maximum values another method using Spark SQL
athletes_table = athletes.createOrReplaceTempView("athletes_table")
max_athletes = spark.sql("""SELECT Country, COUNT(*) AS count FROM athletes_table 
                GROUP BY Country 
                ORDER BY count DESC
                LIMIT 5
                """)
max_athletes.show()

# Getting bottom 5 minimum number of athletes using Spark SQL
min_athletes = spark.sql("""SELECT Country, COUNT(*) AS count FROM athletes_table 
                GROUP BY Country 
                ORDER BY count ASC
                LIMIT 5
                """)
min_athletes.show()

print("As you can see the max number of athletes were hailing from United States, while the minimum number of athletes were hailing from multiple countries such as Myanmar, Tuvalu, and Somalia among many others.")

+--------------------+-----+
|             Country|count|
+--------------------+-----+
|United States of ...|  615|
|               Japan|  586|
|           Australia|  470|
|People's Republic...|  401|
|             Germany|  400|
+--------------------+-----+

+-----------------+-----+
|          Country|count|
+-----------------+-----+
|          Myanmar|    2|
|           Tuvalu|    2|
|Brunei Darussalam|    2|
|         Dominica|    2|
|          Somalia|    2|
+-----------------+-----+

As you can see the max number of athletes were hailing from United States, while the minimum number of athletes were hailing from multiple countries such as Myanmar, Tuvalu, and Somalia among many others.


As you can see the max number of athletes were hailing from United States, while the minimum number of athletes were hailing from multiple countries such as Myanmar, Tuvalu, and Somalia among many others.

### Gender Data

In [0]:
#Displaying Gender percentages from entries_gender data as 2 new columns
from pyspark.sql.functions import round

percentage_table = entries_gender.withColumn("Male Percentage", round(entries_gender["Male"]/entries_gender["Total"]*100,2)) \
                                 .withColumn("Female Percentage", round(entries_gender["Female"]/ entries_gender["Total"]*100,2))
percentage_table.show()

+--------------------+------+----+-----+---------------+-----------------+
|          Discipline|Female|Male|Total|Male Percentage|Female Percentage|
+--------------------+------+----+-----+---------------+-----------------+
|      3x3 Basketball|    32|  32|   64|           50.0|             50.0|
|             Archery|    64|  64|  128|           50.0|             50.0|
| Artistic Gymnastics|    98|  98|  196|           50.0|             50.0|
|   Artistic Swimming|   105|   0|  105|            0.0|            100.0|
|           Athletics|   969|1072| 2041|          52.52|            47.48|
|           Badminton|    86|  87|  173|          50.29|            49.71|
|   Baseball/Softball|    90| 144|  234|          61.54|            38.46|
|          Basketball|   144| 144|  288|           50.0|             50.0|
|    Beach Volleyball|    48|  48|   96|           50.0|             50.0|
|              Boxing|   102| 187|  289|          64.71|            35.29|
|        Canoe Slalom|   

In [0]:
# Getting maximum and minimum participants from the array of Disciplines
from pyspark.sql.functions import max, min
# Maximum number of participants is in the Athletics division, with 1072 Male and 1068 Female athletes.
print("Maximum number of participants is in the Athletics division, with 1072 Male and 1068 Female athletes.")
entries_gender.select("Discipline","Male","Female","Total") \
              .orderBy("Total",ascending=False) \
              .limit(1).show()
# Minimum number of participants is in the Athletics division, with 1072 Male and 1068 Female athletes.
print("Minimum number of participants is in the Athletics division, with 9 Male and 10 Female athletes.")
entries_gender.select("Discipline","Male","Female","Total") \
              .orderBy("Total",ascending=True) \
              .limit(1).show()

#note more effective aggregation functions can be used, the above was for ease of display. 

Maximum number of participants is in the Athletics division, with 1072 Male and 1068 Female athletes.
+----------+----+------+-----+
|Discipline|Male|Female|Total|
+----------+----+------+-----+
| Athletics|1072|   969| 2041|
+----------+----+------+-----+

Minimum number of participants is in the Athletics division, with 9 Male and 10 Female athletes.
+--------------------+----+------+-----+
|          Discipline|Male|Female|Total|
+--------------------+----+------+-----+
|Cycling BMX Frees...|   9|    10|   19|
+--------------------+----+------+-----+



### Teams

In [0]:
# Identifying how many teams per Discipline each Country had
teams.groupBy("Country","Discipline").count().orderBy(["count"],ascending=[False]).show()

+--------------------+-------------+-----+
|             Country|   Discipline|count|
+--------------------+-------------+-----+
|           Australia|     Swimming|    7|
|               Japan|     Swimming|    7|
|United States of ...|     Swimming|    7|
|             Germany|     Swimming|    7|
|                 ROC|     Swimming|    7|
|             Germany|Cycling Track|    6|
|               Italy|      Fencing|    6|
|People's Republic...|     Swimming|    6|
|               Italy|     Swimming|    6|
|              Brazil|     Swimming|    6|
|       Great Britain|     Swimming|    6|
|              Canada|     Swimming|    6|
|                 ROC|      Fencing|    6|
|              Poland|     Swimming|    6|
|United States of ...|      Fencing|    6|
|         New Zealand|Cycling Track|    5|
|       Great Britain|    Athletics|    5|
|             Hungary|     Swimming|    5|
|               Italy|    Athletics|    5|
|       Great Britain|Cycling Track|    5|
+----------

In [0]:
# Identifying most teams per Discipline.
teams.groupBy("Discipline").count() \
                           .withColumnRenamed("count","Team Count") \
                           .orderBy(["count"],ascending=[False]) \
                           .limit(1) \
                           .show()

# Identifying least teams per Discipline.
teams.groupBy("Discipline").count() \
                           .withColumnRenamed("count","Team Count") \
                           .orderBy(["count"],ascending=[True]) \
                           .limit(1) \
                           .show()


+----------+----------+
|Discipline|Team Count|
+----------+----------+
|  Swimming|       113|
+----------+----------+

+-----------------+----------+
|       Discipline|Team Count|
+-----------------+----------+
|Baseball/Softball|        12|
+-----------------+----------+



As shown, the most amount of teams was in the Aquatics Discipline with a total of 113 teams, while the least amount of teams was in the Baseball/Softball Discipline for a total of 12 teams. Popularity of disciplines can be extrapolated from this as a result, with Swimming being the most popular, and Baseball/softball being the least!

### Medals

In [0]:
medals.show()

+----+--------------------+----+------+------+-----+-------------+
|Rank|         TeamCountry|Gold|Silver|Bronze|Total|Rank by Total|
+----+--------------------+----+------+------+-----+-------------+
|   1|United States of ...|  39|    41|    33|  113|            1|
|   2|People's Republic...|  38|    32|    18|   88|            2|
|   3|               Japan|  27|    14|    17|   58|            5|
|   4|       Great Britain|  22|    21|    22|   65|            4|
|   5|                 ROC|  20|    28|    23|   71|            3|
|   6|           Australia|  17|     7|    22|   46|            6|
|   7|         Netherlands|  10|    12|    14|   36|            9|
|   8|              France|  10|    12|    11|   33|           10|
|   9|             Germany|  10|    11|    16|   37|            8|
|  10|               Italy|  10|    10|    20|   40|            7|
|  11|              Canada|   7|     6|    11|   24|           11|
|  12|              Brazil|   7|     6|     8|   21|          

In [0]:

# Finding percentage of Gold medals from total medals for each country
import pyspark.sql.functions as F

total_medals = medals.agg(F.sum("Total")).collect()[0][0]
print("Total medals handed out in the 2021 Tokyo Olympics =", total_medals)
print("-----------------------------------------------------------------")

print("Percentage of Gold Medals Won")
gold_percentage = medals.withColumn(
    "Gold Medals Won", 
    F.round(medals["Gold"] / total_medals * 100, 2)
)
display(gold_percentage)

print("Percentage of Silver Medals Won")
silver_percentage = medals.withColumn(
    "Silver Medals Won", 
    F.round(medals["Silver"] / total_medals * 100, 2)
)
display(silver_percentage)

print("Percentage of Bronze Medals Won")
bronze_percentage = medals.withColumn(
    "Bronze Medals Won", 
    F.round(medals["Bronze"] / total_medals * 100, 2)
)
display(bronze_percentage)

print("Percentage of Total Medals Won")
medal_percentage = medals.withColumn(
    "% Medals Won", 
    F.round(medals["Total"] / total_medals * 100, 2)
)
display(medal_percentage)




Total medals handed out in the 2021 Tokyo Olympics = 1080
-----------------------------------------------------------------
Percentage of Gold Medals Won


Rank TeamCountry Gold Silver Bronze Total Rank by Total Gold Medals Won 1 United States of America 39 41 33 113 1 3.61 2 People's Republic of China 38 32 18 88 2 3.52 3 Japan 27 14 17 58 5 2.5 4 Great Britain 22 21 22 65 4 2.04 5 ROC 20 28 23 71 3 1.85 6 Australia 17 7 22 46 6 1.57 7 Netherlands 10 12 14 36 9 0.93 8 France 10 12 11 33 10 0.93 9 Germany 10 11 16 37 8 0.93 10 Italy 10 10 20 40 7 0.93 11 Canada 7 6 11 24 11 0.65 12 Brazil 7 6 8 21 12 0.65 13 New Zealand 7 6 7 20 13 0.65 14 Cuba 7 3 5 15 18 0.65 15 Hungary 6 7 7 20 13 0.56 16 Republic of Korea 6 4 10 20 13 0.56 17 Poland 4 5 5 14 19 0.37 18 Czech Republic 4 4 3 11 23 0.37 19 Kenya 4 4 2 10 25 0.37 20 Norway 4 2 2 8 29 0.37 21 Jamaica 4 1 4 9 26 0.37 22 Spain 3 8 6 17 17 0.28 23 Sweden 3 6 0 9 26 0.28 24 Switzerland 3 4 6 13 20 0.28 25 Denmark 3 4 4 11 23 0.28 26 Croatia 3 3 2 8 29 0.28 27 Islamic Republic of Iran 3 2 2 7 33 0.28 28 Serbia 3 1 5 9 26 0.28 29 Belgium 3 1 3 7 33 0.28 30 Bulgaria 3 1 2 6 39 0.28 31 Slovenia 3 1 1 5 42 0.28 32 Uzbekistan 3 0 2 5 42 0.28 33 Georgia 2 5 1 8 29 0.19 34 Chinese Taipei 2 4 6 12 22 0.19 35 Turkey 2 2 9 13 20 0.19 36 Greece 2 1 1 4 47 0.19 36 Uganda 2 1 1 4 47 0.19 38 Ecuador 2 1 0 3 60 0.19 39 Ireland 2 0 2 4 47 0.19 39 Israel 2 0 2 4 47 0.19 41 Qatar 2 0 1 3 60 0.19 42 Bahamas 2 0 0 2 66 0.19 42 Kosovo 2 0 0 2 66 0.19 44 Ukraine 1 6 12 19 16 0.09 45 Belarus 1 3 3 7 33 0.09 46 Romania 1 3 0 4 47 0.09 46 Venezuela 1 3 0 4 47 0.09 48 India 1 2 4 7 33 0.09 49 Hong Kong, China 1 2 3 6 39 0.09 50 Philippines 1 2 1 4 47 0.09 50 Slovakia 1 2 1 4 47 0.09 52 South Africa 1 2 0 3 60 0.09 53 Austria 1 1 5 7 33 0.09 54 Egypt 1 1 4 6 39 0.09 55 Indonesia 1 1 3 5 42 0.09 56 Ethiopia 1 1 2 4 47 0.09 56 Portugal 1 1 2 4 47 0.09 58 Tunisia 1 1 0 2 66 0.09 59 Estonia 1 0 1 2 66 0.09 59 Fiji 1 0 1 2 66 0.09 59 Latvia 1 0 1 2 66 0.09 59 Thailand 1 0 1 2 66 0.09 63 Bermuda 1 0 0 1 77 0.09 63 Morocco 1 0 0 1 77 0.09 63 Puerto Rico 1 0 0 1 77 0.09 66 Colombia 0 4 1 5 42 0.0 67 Azerbaijan 0 3 4 7 33 0.0 68 Dominican Republic 0 3 2 5 42 0.0 69 Armenia 0 2 2 4 47 0.0 70 Kyrgyzstan 0 2 1 3 60 0.0 71 Mongolia 0 1 3 4 47 0.0 72 Argentina 0 1 2 3 60 0.0 72 San Marino 0 1 2 3 60 0.0 74 Jordan 0 1 1 2 66 0.0 74 Malaysia 0 1 1 2 66 0.0 74 Nigeria 0 1 1 2 66 0.0 77 Bahrain 0 1 0 1 77 0.0 77 Saudi Arabia 0 1 0 1 77 0.0 77 Lithuania 0 1 0 1 77 0.0 77 North Macedonia 0 1 0 1 77 0.0 77 Namibia 0 1 0 1 77 0.0 77 Turkmenistan 0 1 0 1 77 0.0 83 Kazakhstan 0 0 8 8 29 0.0 84 Mexico 0 0 4 4 47 0.0 85 Finland 0 0 2 2 66 0.0 86 Botswana 0 0 1 1 77 0.0 86 Burkina Faso 0 0 1 1 77 0.0 86 Côte d'Ivoire 0 0 1 1 77 0.0 86 Ghana 0 0 1 1 77 0.0 86 Grenada 0 0 1 1 77 0.0 86 Kuwait 0 0 1 1 77 0.0 86 Republic of Moldova 0 0 1 1 77 0.0 86 Syrian Arab Republic 0 0 1 1 77 0.0

Percentage of Silver Medals Won


Rank TeamCountry Gold Silver Bronze Total Rank by Total Silver Medals Won 1 United States of America 39 41 33 113 1 3.8 2 People's Republic of China 38 32 18 88 2 2.96 3 Japan 27 14 17 58 5 1.3 4 Great Britain 22 21 22 65 4 1.94 5 ROC 20 28 23 71 3 2.59 6 Australia 17 7 22 46 6 0.65 7 Netherlands 10 12 14 36 9 1.11 8 France 10 12 11 33 10 1.11 9 Germany 10 11 16 37 8 1.02 10 Italy 10 10 20 40 7 0.93 11 Canada 7 6 11 24 11 0.56 12 Brazil 7 6 8 21 12 0.56 13 New Zealand 7 6 7 20 13 0.56 14 Cuba 7 3 5 15 18 0.28 15 Hungary 6 7 7 20 13 0.65 16 Republic of Korea 6 4 10 20 13 0.37 17 Poland 4 5 5 14 19 0.46 18 Czech Republic 4 4 3 11 23 0.37 19 Kenya 4 4 2 10 25 0.37 20 Norway 4 2 2 8 29 0.19 21 Jamaica 4 1 4 9 26 0.09 22 Spain 3 8 6 17 17 0.74 23 Sweden 3 6 0 9 26 0.56 24 Switzerland 3 4 6 13 20 0.37 25 Denmark 3 4 4 11 23 0.37 26 Croatia 3 3 2 8 29 0.28 27 Islamic Republic of Iran 3 2 2 7 33 0.19 28 Serbia 3 1 5 9 26 0.09 29 Belgium 3 1 3 7 33 0.09 30 Bulgaria 3 1 2 6 39 0.09 31 Slovenia 3 1 1 5 42 0.09 32 Uzbekistan 3 0 2 5 42 0.0 33 Georgia 2 5 1 8 29 0.46 34 Chinese Taipei 2 4 6 12 22 0.37 35 Turkey 2 2 9 13 20 0.19 36 Greece 2 1 1 4 47 0.09 36 Uganda 2 1 1 4 47 0.09 38 Ecuador 2 1 0 3 60 0.09 39 Ireland 2 0 2 4 47 0.0 39 Israel 2 0 2 4 47 0.0 41 Qatar 2 0 1 3 60 0.0 42 Bahamas 2 0 0 2 66 0.0 42 Kosovo 2 0 0 2 66 0.0 44 Ukraine 1 6 12 19 16 0.56 45 Belarus 1 3 3 7 33 0.28 46 Romania 1 3 0 4 47 0.28 46 Venezuela 1 3 0 4 47 0.28 48 India 1 2 4 7 33 0.19 49 Hong Kong, China 1 2 3 6 39 0.19 50 Philippines 1 2 1 4 47 0.19 50 Slovakia 1 2 1 4 47 0.19 52 South Africa 1 2 0 3 60 0.19 53 Austria 1 1 5 7 33 0.09 54 Egypt 1 1 4 6 39 0.09 55 Indonesia 1 1 3 5 42 0.09 56 Ethiopia 1 1 2 4 47 0.09 56 Portugal 1 1 2 4 47 0.09 58 Tunisia 1 1 0 2 66 0.09 59 Estonia 1 0 1 2 66 0.0 59 Fiji 1 0 1 2 66 0.0 59 Latvia 1 0 1 2 66 0.0 59 Thailand 1 0 1 2 66 0.0 63 Bermuda 1 0 0 1 77 0.0 63 Morocco 1 0 0 1 77 0.0 63 Puerto Rico 1 0 0 1 77 0.0 66 Colombia 0 4 1 5 42 0.37 67 Azerbaijan 0 3 4 7 33 0.28 68 Dominican Republic 0 3 2 5 42 0.28 69 Armenia 0 2 2 4 47 0.19 70 Kyrgyzstan 0 2 1 3 60 0.19 71 Mongolia 0 1 3 4 47 0.09 72 Argentina 0 1 2 3 60 0.09 72 San Marino 0 1 2 3 60 0.09 74 Jordan 0 1 1 2 66 0.09 74 Malaysia 0 1 1 2 66 0.09 74 Nigeria 0 1 1 2 66 0.09 77 Bahrain 0 1 0 1 77 0.09 77 Saudi Arabia 0 1 0 1 77 0.09 77 Lithuania 0 1 0 1 77 0.09 77 North Macedonia 0 1 0 1 77 0.09 77 Namibia 0 1 0 1 77 0.09 77 Turkmenistan 0 1 0 1 77 0.09 83 Kazakhstan 0 0 8 8 29 0.0 84 Mexico 0 0 4 4 47 0.0 85 Finland 0 0 2 2 66 0.0 86 Botswana 0 0 1 1 77 0.0 86 Burkina Faso 0 0 1 1 77 0.0 86 Côte d'Ivoire 0 0 1 1 77 0.0 86 Ghana 0 0 1 1 77 0.0 86 Grenada 0 0 1 1 77 0.0 86 Kuwait 0 0 1 1 77 0.0 86 Republic of Moldova 0 0 1 1 77 0.0 86 Syrian Arab Republic 0 0 1 1 77 0.0

Percentage of Bronze Medals Won


Rank TeamCountry Gold Silver Bronze Total Rank by Total Bronze Medals Won 1 United States of America 39 41 33 113 1 3.06 2 People's Republic of China 38 32 18 88 2 1.67 3 Japan 27 14 17 58 5 1.57 4 Great Britain 22 21 22 65 4 2.04 5 ROC 20 28 23 71 3 2.13 6 Australia 17 7 22 46 6 2.04 7 Netherlands 10 12 14 36 9 1.3 8 France 10 12 11 33 10 1.02 9 Germany 10 11 16 37 8 1.48 10 Italy 10 10 20 40 7 1.85 11 Canada 7 6 11 24 11 1.02 12 Brazil 7 6 8 21 12 0.74 13 New Zealand 7 6 7 20 13 0.65 14 Cuba 7 3 5 15 18 0.46 15 Hungary 6 7 7 20 13 0.65 16 Republic of Korea 6 4 10 20 13 0.93 17 Poland 4 5 5 14 19 0.46 18 Czech Republic 4 4 3 11 23 0.28 19 Kenya 4 4 2 10 25 0.19 20 Norway 4 2 2 8 29 0.19 21 Jamaica 4 1 4 9 26 0.37 22 Spain 3 8 6 17 17 0.56 23 Sweden 3 6 0 9 26 0.0 24 Switzerland 3 4 6 13 20 0.56 25 Denmark 3 4 4 11 23 0.37 26 Croatia 3 3 2 8 29 0.19 27 Islamic Republic of Iran 3 2 2 7 33 0.19 28 Serbia 3 1 5 9 26 0.46 29 Belgium 3 1 3 7 33 0.28 30 Bulgaria 3 1 2 6 39 0.19 31 Slovenia 3 1 1 5 42 0.09 32 Uzbekistan 3 0 2 5 42 0.19 33 Georgia 2 5 1 8 29 0.09 34 Chinese Taipei 2 4 6 12 22 0.56 35 Turkey 2 2 9 13 20 0.83 36 Greece 2 1 1 4 47 0.09 36 Uganda 2 1 1 4 47 0.09 38 Ecuador 2 1 0 3 60 0.0 39 Ireland 2 0 2 4 47 0.19 39 Israel 2 0 2 4 47 0.19 41 Qatar 2 0 1 3 60 0.09 42 Bahamas 2 0 0 2 66 0.0 42 Kosovo 2 0 0 2 66 0.0 44 Ukraine 1 6 12 19 16 1.11 45 Belarus 1 3 3 7 33 0.28 46 Romania 1 3 0 4 47 0.0 46 Venezuela 1 3 0 4 47 0.0 48 India 1 2 4 7 33 0.37 49 Hong Kong, China 1 2 3 6 39 0.28 50 Philippines 1 2 1 4 47 0.09 50 Slovakia 1 2 1 4 47 0.09 52 South Africa 1 2 0 3 60 0.0 53 Austria 1 1 5 7 33 0.46 54 Egypt 1 1 4 6 39 0.37 55 Indonesia 1 1 3 5 42 0.28 56 Ethiopia 1 1 2 4 47 0.19 56 Portugal 1 1 2 4 47 0.19 58 Tunisia 1 1 0 2 66 0.0 59 Estonia 1 0 1 2 66 0.09 59 Fiji 1 0 1 2 66 0.09 59 Latvia 1 0 1 2 66 0.09 59 Thailand 1 0 1 2 66 0.09 63 Bermuda 1 0 0 1 77 0.0 63 Morocco 1 0 0 1 77 0.0 63 Puerto Rico 1 0 0 1 77 0.0 66 Colombia 0 4 1 5 42 0.09 67 Azerbaijan 0 3 4 7 33 0.37 68 Dominican Republic 0 3 2 5 42 0.19 69 Armenia 0 2 2 4 47 0.19 70 Kyrgyzstan 0 2 1 3 60 0.09 71 Mongolia 0 1 3 4 47 0.28 72 Argentina 0 1 2 3 60 0.19 72 San Marino 0 1 2 3 60 0.19 74 Jordan 0 1 1 2 66 0.09 74 Malaysia 0 1 1 2 66 0.09 74 Nigeria 0 1 1 2 66 0.09 77 Bahrain 0 1 0 1 77 0.0 77 Saudi Arabia 0 1 0 1 77 0.0 77 Lithuania 0 1 0 1 77 0.0 77 North Macedonia 0 1 0 1 77 0.0 77 Namibia 0 1 0 1 77 0.0 77 Turkmenistan 0 1 0 1 77 0.0 83 Kazakhstan 0 0 8 8 29 0.74 84 Mexico 0 0 4 4 47 0.37 85 Finland 0 0 2 2 66 0.19 86 Botswana 0 0 1 1 77 0.09 86 Burkina Faso 0 0 1 1 77 0.09 86 Côte d'Ivoire 0 0 1 1 77 0.09 86 Ghana 0 0 1 1 77 0.09 86 Grenada 0 0 1 1 77 0.09 86 Kuwait 0 0 1 1 77 0.09 86 Republic of Moldova 0 0 1 1 77 0.09 86 Syrian Arab Republic 0 0 1 1 77 0.09

Percentage of Total Medals Won


Rank TeamCountry Gold Silver Bronze Total Rank by Total % Medals Won 1 United States of America 39 41 33 113 1 10.46 2 People's Republic of China 38 32 18 88 2 8.15 3 Japan 27 14 17 58 5 5.37 4 Great Britain 22 21 22 65 4 6.02 5 ROC 20 28 23 71 3 6.57 6 Australia 17 7 22 46 6 4.26 7 Netherlands 10 12 14 36 9 3.33 8 France 10 12 11 33 10 3.06 9 Germany 10 11 16 37 8 3.43 10 Italy 10 10 20 40 7 3.7 11 Canada 7 6 11 24 11 2.22 12 Brazil 7 6 8 21 12 1.94 13 New Zealand 7 6 7 20 13 1.85 14 Cuba 7 3 5 15 18 1.39 15 Hungary 6 7 7 20 13 1.85 16 Republic of Korea 6 4 10 20 13 1.85 17 Poland 4 5 5 14 19 1.3 18 Czech Republic 4 4 3 11 23 1.02 19 Kenya 4 4 2 10 25 0.93 20 Norway 4 2 2 8 29 0.74 21 Jamaica 4 1 4 9 26 0.83 22 Spain 3 8 6 17 17 1.57 23 Sweden 3 6 0 9 26 0.83 24 Switzerland 3 4 6 13 20 1.2 25 Denmark 3 4 4 11 23 1.02 26 Croatia 3 3 2 8 29 0.74 27 Islamic Republic of Iran 3 2 2 7 33 0.65 28 Serbia 3 1 5 9 26 0.83 29 Belgium 3 1 3 7 33 0.65 30 Bulgaria 3 1 2 6 39 0.56 31 Slovenia 3 1 1 5 42 0.46 32 Uzbekistan 3 0 2 5 42 0.46 33 Georgia 2 5 1 8 29 0.74 34 Chinese Taipei 2 4 6 12 22 1.11 35 Turkey 2 2 9 13 20 1.2 36 Greece 2 1 1 4 47 0.37 36 Uganda 2 1 1 4 47 0.37 38 Ecuador 2 1 0 3 60 0.28 39 Ireland 2 0 2 4 47 0.37 39 Israel 2 0 2 4 47 0.37 41 Qatar 2 0 1 3 60 0.28 42 Bahamas 2 0 0 2 66 0.19 42 Kosovo 2 0 0 2 66 0.19 44 Ukraine 1 6 12 19 16 1.76 45 Belarus 1 3 3 7 33 0.65 46 Romania 1 3 0 4 47 0.37 46 Venezuela 1 3 0 4 47 0.37 48 India 1 2 4 7 33 0.65 49 Hong Kong, China 1 2 3 6 39 0.56 50 Philippines 1 2 1 4 47 0.37 50 Slovakia 1 2 1 4 47 0.37 52 South Africa 1 2 0 3 60 0.28 53 Austria 1 1 5 7 33 0.65 54 Egypt 1 1 4 6 39 0.56 55 Indonesia 1 1 3 5 42 0.46 56 Ethiopia 1 1 2 4 47 0.37 56 Portugal 1 1 2 4 47 0.37 58 Tunisia 1 1 0 2 66 0.19 59 Estonia 1 0 1 2 66 0.19 59 Fiji 1 0 1 2 66 0.19 59 Latvia 1 0 1 2 66 0.19 59 Thailand 1 0 1 2 66 0.19 63 Bermuda 1 0 0 1 77 0.09 63 Morocco 1 0 0 1 77 0.09 63 Puerto Rico 1 0 0 1 77 0.09 66 Colombia 0 4 1 5 42 0.46 67 Azerbaijan 0 3 4 7 33 0.65 68 Dominican Republic 0 3 2 5 42 0.46 69 Armenia 0 2 2 4 47 0.37 70 Kyrgyzstan 0 2 1 3 60 0.28 71 Mongolia 0 1 3 4 47 0.37 72 Argentina 0 1 2 3 60 0.28 72 San Marino 0 1 2 3 60 0.28 74 Jordan 0 1 1 2 66 0.19 74 Malaysia 0 1 1 2 66 0.19 74 Nigeria 0 1 1 2 66 0.19 77 Bahrain 0 1 0 1 77 0.09 77 Saudi Arabia 0 1 0 1 77 0.09 77 Lithuania 0 1 0 1 77 0.09 77 North Macedonia 0 1 0 1 77 0.09 77 Namibia 0 1 0 1 77 0.09 77 Turkmenistan 0 1 0 1 77 0.09 83 Kazakhstan 0 0 8 8 29 0.74 84 Mexico 0 0 4 4 47 0.37 85 Finland 0 0 2 2 66 0.19 86 Botswana 0 0 1 1 77 0.09 86 Burkina Faso 0 0 1 1 77 0.09 86 Côte d'Ivoire 0 0 1 1 77 0.09 86 Ghana 0 0 1 1 77 0.09 86 Grenada 0 0 1 1 77 0.09 86 Kuwait 0 0 1 1 77 0.09 86 Republic of Moldova 0 0 1 1 77 0.09 86 Syrian Arab Republic 0 0 1 1 77 0.09

As displayed above, the percentage of Gold, Silver and Bronze medals can be seen, as well as the total medals won by each country as a percentage from the total medals handed out in the world cup. 

**The Top 3 Countries:**

United States of America took home 10.76% of all medals <br>
People's Republic of China took home 8.15% of all medals <br>
Japan took home 5.37% of all medals <br>

Writing Data to output location in Azure Data Lake Storage:

In [0]:
# Writing default raw data tables that were loaded.
athletes.write.option("header","true").csv("dbfs:/mnt/tokyoolympic/transformed-data/athletes")
coaches.write.option("header","true").csv("dbfs:/mnt/tokyoolympic/transformed-data/coaches")
entries_gender.write.option("header","true").csv("dbfs:/mnt/tokyoolympic/transformed-data/entries_gender")
medals.write.option("header","true").csv("dbfs:/mnt/tokyoolympic/transformed-data/medals")
teams.write.option("header","true").csv("dbfs:/mnt/tokyoolympic/transformed-data/teams")


In [0]:
# Writing custom transformed tables that provided detailed, specific insights.
basketball.write.option("header","true").csv("dbfs:/mnt/tokyoolympic/transformed-data/basketball")
sport_popularity.write.option("header","true").csv("dbfs:/mnt/tokyoolympic/transformed-data/sport_popularity")
max_athletes.write.option("header","true").csv("dbfs:/mnt/tokyoolympic/transformed-data/max_athletes")
min_athletes.write.option("header","true").csv("dbfs:/mnt/tokyoolympic/transformed-data/min_athletes")
gold_percentage.write.option("header","true").csv("dbfs:/mnt/tokyoolympic/transformed-data/gold_percentage")
silver_percentage.write.option("header","true").csv("dbfs:/mnt/tokyoolympic/transformed-data/silver_percentage")
bronze_percentage.write.option("header","true").csv("dbfs:/mnt/tokyoolympic/transformed-data/bronze_percentage")
medal_percentage.write.option("header","true").csv("dbfs:/mnt/tokyoolympic/transformed-data/medal_percentage")